### PR TITLE
Mining asteroids overhaul

### PR DESCRIPTION
**Pull request by Vonsant**
## About the PR

Just added a little variety to the diggers' routine. From my Corvax Frontier fork, people loves this.
https://github.com/Corvax-Frontier/corvax-frontier-14/pull/151

## Why / Balance

Standard asteroids are boring. Kangaroos, bears and carps are already in everyone's throat.
## How to test

Start a game.
Fly through the space.
Find asteroid.
Diggy Diggy Hole.
Kill some crabs.
Spawn some rock anomaly.
Kill some crabs.

## Media

image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Nothing, i think.

**Changelog**

Added new ore crabs for all the missing types of resources in asteroids.
Added new rock anomaly.
Added rare ore golems.
Added crabs and golems in every goddamn WallRock
Added crabs and golems spawn
Added several different asteroids with different ore spawn chance.
Added more asteroids type to spawn in space.

### CODE REVIEW

**Comment by dvir001** [comment]
Nice, Ill try and get it organized into _NF if you need help

**Comment by Vonsant** [comment]
> Nice, Ill try and get it organized into _NF if you need help

If you think it will the best option, I have nothing against.
I'm just update files right now, so its not everything done right now/

**Comment by Vonsant** [comment]
> Nice, Ill try and get it organized into _NF if you need help

Welp, i've did my best. Do you wanna change something?

**Comment by Vonsant** [comment]
mimicgif
68d16e6ca446f77b99e447fc2bc3ed3f

Maybe next time it will be Wreck update. Maybe... Sometimes... When I can make upstream... There is no more Bears in space... No one hear your scream, when you find not just a regular crap on wreck.

**Comment by Vonsant** [comment]
ffs

**Comment by Vonsant** [comment]
Ok, now it's done.

**Comment by Cheackraze** [comment]
nice, theyve needed some love for a long time

**Review by Cheackraze — CHANGES_REQUESTED** [review]
everything looks fine, just need everything moved to _NF/ folders

**Comment by Vonsant** [comment]
I think it's done now?

**Comment by dvir001** [comment]
I'll have a look this week, going to clean some issues with the code before we add more stuff

Looking great 👍

**Review by Cheackraze — APPROVED** [review]
looks good for the most part, pretty much all yml. If one of the other maintainers doesnt give it a test run, I will sometime this week and we will get this in.

**Comment by dvir001** [comment]
> looks good for the most part, pretty much all yml. If one of the other maintainers doesnt give it a test run, I will sometime this week and we will get this in.

Wait a bit we want to clean the yml

**Review by whatston3 — COMMENTED** [review]
Pretty excited for this, looks like an excellent feature - I think the changes would help give variety and some additional needed lethality to salvage gameplay and should give medics a good bit more work.  The variety of asteroid types with different resource distributions is also great and allows for science vessel pilots to pick and choose when short on a resource.

My only gripe with this so far is that it's hard to tell when the golems are dead.  Even an edited, darker sprite as a placeholder would make threat assessment easier.

**Comment by dvir001** [comment]
It is done.

File: Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+            orGroup: rock
+          - id: WallRockBasaltCoal
+            prob: 0.01
+            orGroup: rock
+          - id: WallRockBasaltArtifactFragment
+            prob: 0.01
+            orGroup: rock
+          - id: WallBasaltCobblebrick
+            prob: 0.1
+            orGroup: rock
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml:524]:
```suggestion
            orGroup: rock
    - type: IFF
      flags: HideLabel
      color: "#b5b5b5"
```

Basalt rocks currently inherit their map colour from the base.  This suggested colour is a grey, to match the grey of the rock.  How this looks in-game:
image
The rock is readable against the UI rings and the nearby Pioneer's outline.

File: Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml [hunk 2]
```diff
@@ -0,0 +1,1078 @@
...
+          - id: WallRockChromiteSilver
+            prob: 0.01
+            orGroup: rock
+          - id: WallChromiteCobblebrick
+            prob: 0.1
+            orGroup: rock
+    - type: IFF
+      flags: HideLabel
+      color: "#7d75bf"
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml:832]:
```suggestion
      color: "#8178cc"
```

Chromite asteroids show up as a bit dark on the radar.  This makes them slightly easier to read.  In-game:
image
Should better stand out against the background and be readily distinguishable from wrecks.

**Review by whatston3 — COMMENTED** [review]
Looks good so far.  Haven't reviewed the latest changes in their entirety, but a few comments and suggestions from what I've seen.

File: Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml [hunk 3]
```diff
@@ -0,0 +1,1078 @@
...
+            prob: 0.01
+            orGroup: rock
+          - id: WallNecropolis
+            prob: 0.1
+            orGroup: rock
+    - type: GCAbleObject
+      queue: SpaceDebris
+    - type: IFF
+      flags: HideLabel
+      color: "#d67e27"
```
[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Entities/World/Debris/asteroids.yml:988]: I don't think I've seen an instance of this asteroid type spawned.  I'll look into it a bit more when I can.

**Comment by Cheackraze** [comment]
I agree with whatson, and if this is ready I'd like to get it in for the weekend

**Comment by dvir001** [comment]
> I agree with whatson, and if this is ready I'd like to get it in for the weekend

You can approve the segested changes, I won't have time to test it again today.

**Pull request merged**